phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -1094,6 +1095,11 @@ private void addCoprocessors(byte[] tableName, TableDescriptorBuilder builder,
                 if(!newDesc.hasCoprocessor(TaskRegionObserver.class.getName())) {
                     builder.addCoprocessor(TaskRegionObserver.class.getName(), null, priority, null);
                 }
+                if (!newDesc.hasCoprocessor(
```

Comment:
If SYSTEM.TASK already exists on a cluster and you upgrade to 4.16 server bits and connect with a 4.16 client, will we still install this new coproc or does that need extra steps in the upgrade path?

Reply:
Yes, we would need to take care of upgrade path. Updated PR with changes in `upgradeSystemTask()` function by handling addition of this new coprocessor, test is available in `BackwardCompatibilityIT` for existence of new coprocessor in updated table descriptor.
Thanks for the suggestion @ChinmaySKulkarni